File: live-android-native/app/src/main/java/com/haoke91/a91edu/widget/NoScrollViewPager.java
package com.haoke91.a91edu.widget;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;

/**
 * 项目名称：91HaoKe_Android
 * 类描述：
 * 创建人：shichengxiang
 * 创建时间：2018/7/12 10:33
 */
public class NoScrollViewPager extends ViewPager {
    public NoScrollViewPager(@NonNull Context context) {
        super(context);
    }
    
    public NoScrollViewPager(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }
    
    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return false;
    }
    
    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return false;
    }
    
    @Override
    public void setCurrentItem(int item) {
        super.setCurrentItem(item, false);
    }
    /**/
}

File: app-student-android/GStudentLib/src/main/java/com/gstudentlib/view/viewpager/NoScrollViewPager.kt
package com.gstudentlib.view.viewpager

import android.content.Context
import android.support.v4.view.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent

/**
 * 作者：created by 逢二进一 on 2019/9/17 11:16
 * 邮箱：dingyuanzheng@gaosiedu.com
 */
class NoScrollViewPager : ViewPager {

    private var isScrollable = false

    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    fun setScrollable(scrollable: Boolean) {
        isScrollable = scrollable
    }

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        try {
            return isScrollable && super.onTouchEvent(ev)
        } catch (e: IllegalArgumentException) {
            e.printStackTrace()
        }
        return false
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        try {
            return isScrollable && super.onInterceptTouchEvent(ev)
        } catch (e: IllegalArgumentException) {
            e.printStackTrace()
        }
        return false
    }

}